search/median_search2.cpp

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file
+ *
+ * @details
+ * Given a linked list L[0,....,n] of n numbers, find the middle node. 
+ * The technique utilized in this implementation is the "Floyd's tortise and hare" approach. Wikipedia link to technique: https://en.wikipedia.org/wiki/Cycle_detection#Floyd's_tortoise_and_hare
+ * This technique uses two pointers that iterate through the list at different 'speeds' in order to solve problems.
+ * In this implementation, for every iteration the slow pointer advances one node while the fast pointer advances two nodes. 
+ * The result of this is that since the fast pointer moves twice as fast as the slow pointer, when the fast pointer reaches the end of the list 
+ * the slow pointer will be pointing to the middle node of the list. 
+ *
+ * Here are some example lists you can use to see how the algorithm works
+ * A = [1,2,3,4,5]
+ * B = [1,2,3,4,5,6]
+ * print median(A) #should be 39
+ * print median(B) #should be 4
+ *
+ * @author [Benjamin Weiss](https://github.com/weiss-ben)
+ */
+
+/**
+ * Definition for singly-linked list.
+ */
+struct ListNode {
+    int val;												// The value stored in the node
+    ListNode *next;											// Pointer to the next node
+    ListNode() : val(0), next(nullptr) {}					// Default constructor
+    ListNode(int x) : val(x), next(nullptr) {}				// Constructor with value for node->val provided
+    ListNode(int x, ListNode *next) : val(x), next(next) {}	// Constructor with values provided for node->val and node->next
```

Comment:
```suggestion
    int val;                                                   ///< the value stored in the node
    ListNode *next;                                            ///< pointer to the next node
    ListNode() : val(0), next(nullptr) {}                     ///< default constructor
    ListNode(int x) : val(x), next(nullptr) {}               ///< constructor with value for node->val provided
    ListNode(int x, ListNode *next) : val(x), next(next) {}  ///< constructor with values provided for node->val and node->next
```


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Inelegant